### PR TITLE
Integration test feedback

### PR DESCRIPTION
**Pull request by beccaelenzil**
This PR is not intended to merged into AdaGold. It is meant as a place to provide feedback on the integration tests. 

There is a full solution provided in `party.py`

I drafted integration tests -- 1 per wave. Some of these are not true integration tests, but more unit tests strung together, as there is not much integration to test in every wave.

Questions: 
1. Should we combine the some of the waves into a single integration test so that we're able to test more of the integration?
2. Should we be testing more edge cases in the integration tests?

I also created a `tests_constant.py` file. Rather than giving each function `user_data` that's differently shaped for each wave/function as the requirements in the readme currently describe, the `USER_DATA` in `tests_constants.py` builds in complexity (by adding keys) with each wave. Similarly, keys are added to the movie dictionaries as needed.

Questions:
1. Should we use these `tests_constants` or similar in the unit tests?
2. Should we update the readme to describe the `user_data` and `movie`data as building in complexity (adding keys each wave) as is done with the `USER_DATA` in `test_constants.py`?

### CODE REVIEW

File: tests/integration_tests/test_integration_wave_02.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+def test_get_most_watched_genre():
+    # Arrange
+    user_data = USER_DATA
+    empty_user_data = EMPTY_USER_DATA
+
+    # user data
+    most_watched = get_most_watched_genre(user_data)
+    average_rating = get_watched_avg_rating(user_data)
+    assert most_watched == "Fantasy"
+    assert average_rating == pytest.approx(3.58333)
```
Comment (kaidamasaki) [inline tests/integration_tests/test_integration_wave_02.py:15]: Good catch using approx!

File: tests/test_constants.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    "title": "Intrigue 2",
+    "genre": "Intrigue",
+    "rating": 4.5
+}
+
+INTRIGUE_3 = {
+    "title": "Intrigue 3",
+    "genre": "Intrigue",
+    "rating": 3.0
+}
```
Comment (kaidamasaki) [inline tests/test_constants.py:71]: More fun names:

```suggestion
MOVIE_TITLE_1 = "It Came from the Stack Trace"
GENRE_1 = "Horror"
RATING_1 = 3.5

#----------WAVE02-------------
HORROR_1 = {
    "title": MOVIE_TITLE_1,
    "genre": GENRE_1,
    "rating": RATING_1
}

FANTASY_1 = {
    "title": "The Lord of the Functions: The Fellowship of the Function",
    "genre": "Fantasy",
    "rating": 4.8
}

FANTASY_2 = {
    "title": "The Lord of the Functions: The Two Parameters",
    "genre": "Fantasy",
    "rating": 4.0
}

FANTASY_3 = {
    "title": "The Lord of the Functions: The Return of the Value",
    "genre": "Fantasy",
    "rating": 4.0
}

FANTASY_4 = {
    "title": "The Programmer: An Unexpected Stack Trace",
    "genre": "Fantasy",
    "rating": 4.0
}

ACTION_1 = {
    "title": "The JavaScript and the React",
    "genre": "Action",
    "rating": 2.2
}

ACTION_2 = {
    "title": "2 JavaScript 2 React",
    "genre": "Action",
    "rating": 4.2
}

ACTION_3 = {
    "title": "JavaScript 3: VS Code Lint",
    "genre": "Action",
    "rating": 3.5
}

INTRIGUE_1 = {
    "title": "Recursion",
    "genre": "Intrigue",
    "rating": 2.0
}

INTRIGUE_2 = {
    "title": "Instructor Student TA Manager",
    "genre": "Intrigue",
    "rating": 4.5
}

INTRIGUE_3 = {
    "title": "Zero Dark Python",
    "genre": "Intrigue",
    "rating": 3.0
}
```

**Review by kaidamasaki — COMMENTED** [review]
This looks pretty good.

One thing I'd consider doing is adding more tests cases for each wave.  We also might want to change the name from "integration tests" to "smoke tests" since that's a vaguer term.  XD  (Which I think is why we used it with the Postman tests.)
